packages/react/src/components/button.tsx

```diff
@@ -106,128 +104,104 @@ export function UploadButton<
 
   const fileInputRef = useRef<HTMLInputElement>(null);
   const labelRef = useRef<HTMLLabelElement>(null);
-  const [uploadProgressState, setUploadProgress] = useState(
+  const [uploadProgress, setUploadProgress] = useState(
     $props.__internal_upload_progress ?? 0,
   );
-  const [files, setFiles] = useState<File[]>([]);
-  const [isManualTriggerDisplayed, setIsManualTriggerDisplayed] =
-    useState(false);
-  const uploadProgress =
-    $props.__internal_upload_progress ?? uploadProgressState;
```

Comment:
we already pass `__internal_upload_progress` into `setState` so no need to force it here too. We never chagne the progress in docs so doesn't have to be "dynamic"